server/src/main/java/org/elasticsearch/transport/TcpChannel.java

```diff
@@ -79,4 +86,40 @@
      * @param listener to be executed
      */
     void addConnectListener(ActionListener<Void> listener);
+
+    /**
+     * Returns stats about this channel
+     */
+    ChannelStats getChannelStats();
+
+    class ChannelStats {
+
+        private volatile long lastReadTime;
+        private volatile long lastWriteTime;
+
+        public ChannelStats() {
+            // We set the initial value to a day in the past to ensure that the next read or write time is
+            // greater than the initial time. This is important because we mark reads and writes with our
```

Comment:
why does it need to be higher? the comparison here https://github.com/elastic/elasticsearch/pull/35441/files#diff-2dc062611a89dbcea48ead749790ac07R205 is `<= 0` so it's fine to be equal? not sure I follow

Reply:
The cached time updater ticks every 200 ms. So `System.nano()` can return a time that is in the future of the time it returns.

Reply:
I don't understand. We always get the time from the threadpool so how can it be in the future?

Reply:
In the version you reviewed here, we still used `System.nanoTime()` when the `ChannelStats` were ctored. That was to avoid passing the thread pool to each channel impl (even if it was a method ref for `LongSupplier`). And then passing that to the `ChannelStats`.

Since `ChannelStats` are now ctored in `TcpTransport`, it is very easy to pass the thread pool in.